joblib/test/test_parallel.py

```diff
@@ -1225,3 +1226,20 @@ def register_foo():
 
     with parallel_backend('foo'):
         assert isinstance(Parallel()._backend, ThreadingBackend)
+
+
+def _recursive_parallel():
+    # A horrible function that does recursive parallelist
+    Parallel()(delayed(_recursive_parallel)() for i in range(2))
+
+
+def test_fork_bomp():
+    # Test that recursive parallelism raises a recursion rather than
+    # doing a fork bomp
+    # Depending on whether the exception is raised in the main thread
+    # or in a slave thread and the version of Python one exception org
+    # another is raised
+    with parallel_backend('threading', n_jobs=-1):
```

Comment:
It would be even better to test with the default backend: loky would be used as the top level and threads in the nested calls.